src/inmanta/server/services/compilerservice.py

```diff
@@ -645,10 +644,10 @@ async def request_recompile(
         :param in_db_transaction: If set to True, the connection must be provided and the connection must be part of an ongoing
                                   database transaction. If this parameter is set to True, is required to call
                                   `CompileService.notify_compile_request_committed()` right after the transaction commits.
-        :param soft_delete: Silently ignore deletion of resource sets in removed_resource_sets if they contain
-            resources that are being exported.
-        :param mergeable_env_vars: a set of env vars that can be compacted over multiple compiles.
-            If multiple values are compacted, they will be joined using spaces
+        :param soft_delete: Silently ignore deletion of resource sets passed in the env_vars or mergeable_env_vars if
+            they contain resources that are being exported.
+        :param mergeable_env_vars: a set of env vars that can be compacted over multiple compiles when batched partial compiles
```

Comment:
At this level, we don't care about partial compile (i.e mergeable env vars can be used for other things as well)